### DISCUSSION TITLE
Cannot capture Query params using `next/Router`

### DISCUSSION THREAD

**Discussion by Marin-Kitagawa** [vercel/next.js#42085] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
Operating System:

      Platform: win32

      Arch: x64

      Version: Windows 10 Enterprise

    Binaries:

      Node: 16.17.1

      npm: N/A

      Yarn: N/A

      pnpm: N/A

    Relevant packages:

      next: 12.2.3

      eslint-config-next: 12.2.3

      react: 18.2.0

      react-dom: 18.2.0
```

### What browser are you using? (if relevant)

Edge 106.0.1370.34

### How are you deploying your application? (if relevant)

`yarn dev`

### Describe the Bug

I have an API route /api/form
```js
// ./pages/api/form.js
import Router from 'next/router';
...
export default async function handler(req, res) {
    ...
    res.redirect(307, '/summary?username=username');
    Router.push({
        pathname: '/summary',
        query: {
            username: username
        }
    });
}
```

```js
// ./pages/summary.js
import { useRouter } from 'next/router';
export default function Summary() {
    const router = useRouter();
    console.log(router.query);  // {}
}
```

I am not able to fetch the query param. Also, if change the order of `Router.push` and `res.redirect`, I still stay on the /api/form route. Using them individually also didn't solve the problem. Query always comes as `{}`. 

I also tried using `useRouter().push` as per the documentation. Still, I stay in the `/api/form route`.

### Expected Behavior

I must have obtained the query parameter

### Link to reproduction

-

### To Reproduce

Just see the above code

**Comment by MohamedBechirMejri** [comment]
I think this should help you: 
- https://beta.nextjs.org/docs/api-reference/use-search-params
